tools/pyi/gen_pyi.py

```diff
@@ -35,46 +37,32 @@
 read gen_pyi for the gory details.
 """
 
-# TODO: remove after migrating entire codegen to the new data model.
-def should_generate_python_binding(declaration):
-    name = declaration['name']
-    for pattern in SKIP_PYTHON_BINDINGS:
-        if re.match('^' + pattern + '$', name):
-            return False
-
-    simple_types = [arg['simple_type'] for arg in declaration['arguments']]
-    signature = '{}({})'.format(name, ', '.join(simple_types))
-    for pattern in SKIP_PYTHON_BINDINGS_SIGNATURES:
-        if pattern == signature:
-            return False
-
-    return True
-
+# TODO: consider waiting to group by base name until we actually need to (after computing type hint signatures, when adding @overload directives)
+def group_by_base_name(python_funcs: Sequence[PythonSignatureNativeFunctionPair]) -> Dict[str, Sequence[PythonSignatureGroup]]:
+    groups = group_overloads(python_funcs, sort=False)
+    d = collections.defaultdict(list)
+    for g in groups:
+        name = g.signature.name
+        d[name].append(g)
+    return d
 
-def get_py_variable_methods(declarations):
+def get_py_torch_functions(python_funcs: Sequence[PythonSignatureNativeFunctionPair], method: bool = False) -> Dict[str, Sequence[PythonSignatureGroup]]:
     """
     Get declarations (grouped by name) which should be generated
-    as methods on Tensor.
+    as either functions in the "torch" module or methods on Tensor.
     """
-    def should_bind(declaration):
-        return (should_generate_python_binding(declaration) and
-                not declaration.get('python_module') and
-                is_tensor_method(declaration))
+    def should_bind_function(python_func):
+        return (should_generate_py_binding(python_func.function) and
+                not python_func.function.python_module and
+                'namespace' in compute_method_of_yaml(python_func.function.variants))
 
-    return group_declarations_by_op_name([d for d in declarations if should_bind(d)])
-
-
-def get_py_torch_functions(declarations):
-    """
-    Get declarations (grouped by name) which should be generated
-    as functions in the "torch" module.
-    """
-    def should_bind(declaration):
-        return (should_generate_python_binding(declaration) and
-                not declaration.get('python_module') and
-                is_torch_function(declaration))
+    def should_bind_method(python_func):
+        return (should_generate_py_binding(python_func.function) and
+                not python_func.function.python_module and
+                'Tensor' in compute_method_of_yaml(python_func.function.variants))
```

Comment:
Use of `compute_method_of_yaml` here looks super questionable.

Reply:
ha yeah.... fixed.